lib/matplotlib/backends/__init__.py

```diff
@@ -43,50 +44,19 @@ def pylab_setup(name=None):
     '''
     # Import the requested backend into a generic module object
     if name is None:
-        # validates, to match all_backends
         name = matplotlib.get_backend()
-    if name.startswith('module://'):
-        backend_name = name[9:]
-    else:
-        backend_name = 'backend_' + name
-        backend_name = backend_name.lower()  # until we banish mixed case
-        backend_name = 'matplotlib.backends.%s' % backend_name.lower()
-
-    # the last argument is specifies whether to use absolute or relative
-    # imports. 0 means only perform absolute imports.
-    backend_mod = __import__(backend_name, globals(), locals(),
-                             [backend_name], 0)
-
-    # Things we pull in from all backends
-    new_figure_manager = backend_mod.new_figure_manager
-
-    # image backends like pdf, agg or svg do not need to do anything
-    # for "show" or "draw_if_interactive", so if they are not defined
-    # by the backend, just do nothing
-    def do_nothing_show(*args, **kwargs):
-        frame = inspect.currentframe()
-        fname = frame.f_back.f_code.co_filename
-        if fname in ('<stdin>', '<ipython console>'):
-            warnings.warn("""
-Your currently selected backend, '%s' does not support show().
-Please select a GUI backend in your matplotlibrc file ('%s')
-or with matplotlib.use()""" %
-                          (name, matplotlib.matplotlib_fname()))
-
-    def do_nothing(*args, **kwargs):
-        pass
-
-    backend_version = getattr(backend_mod, 'backend_version', 'unknown')
-
-    show = getattr(backend_mod, 'show', do_nothing_show)
-
-    draw_if_interactive = getattr(backend_mod, 'draw_if_interactive',
-                                  do_nothing)
-
-    _log.debug('backend %s version %s', name, backend_version)
+    backend_name = (name[9:] if name.startswith("module://")
+                    else "matplotlib.backends.backend_{}".format(name.lower()))
+    backend_mod = importlib.import_module(backend_name)
+    Backend = type("Backend", (_Backend,), vars(backend_mod))
```

Comment:
Can you add a comment explaining this bit of magic?  I get that it returns a base _Backend class, but am not quite sure what `vars(backend_mod)` does.

Reply:
vars(obj) returns the attributes of obj as a dict.  In the case of a module, that's basically the module's contents.

So effectively it's as if I'm running all of the module's code (of course, not a second time) from within a `class Backend(_Backend): ...` context, and refetching attributes from that class.  The point being that the base _Backend class can provide default implementations of backend_version, show, draw_if_interactive through inheritance rather than pylab_setup having to do it manually.  Will add a comment to that effect, but does that sound good?

Reply:
yeah, not objecting to the magic at all, but I think a short comment to help those of us with lesser python skills know whats going on would help make sure the code stays maintained...

Reply:
added comment.